### PR TITLE
[Python] Add conversation summary skill

### PR DESCRIPTION
**Pull request by kitnhl**
### Motivation and Context
Porting from C#

### Description
The `semantic_text_partitioner` class and `SKFunctionBase.aggregate_partitioned_results_async(...)` method still need to be implemented for the skill to be operational, but for the sake of modularity and PR granularity, I will leave these implementations outside the scope of this particular PR.

### Contribution Checklist

- [x] The code builds clean without any errors or warnings
- [x] The PR follows SK Contribution Guidelines (https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md)
- [x] The code follows the .NET coding conventions (https://learn.microsoft.com/dotnet/csharp/fundamentals/coding-style/coding-conventions) verified with `dotnet format`
- [x] All unit tests pass, and I have added new tests where possible
- [x] I didn't break anyone :smile:

### CODE REVIEW

**Comment by jjhenkel** [comment]
Hey thanks for working on this! I know it's in a draft state, but one bit of early feedback: maybe in the Python version we can put those constants w/ the skill? (This will also help reduce conflicts as we have others working on similar skills that used to, in the C# version, store their prompts in a "constants" file.) 

Open to thoughts/feedback!

**Comment by jjhenkel** [comment]
Oh and as we're evolving things, soon the `sk.extensions.create_*(kernel, ...)` is getting cleaned up to be just `kernel.create_(...)` (so long as you use `Kernel` as the type and not `KernelBase`)

**Comment by kitnhl** [comment]
Hi Jordan! Some of the prompts are really long and I feel like it's less readable if it's just part of the code file, so I still have them in a constants file for now. Please let me know if you feel strongly the other way and I'll be happy to change it.

**Comment by kitnhl** [comment]
> Hey thanks for working on this! I know it's in a draft state, but one bit of early feedback: maybe in the Python version we can put those constants w/ the skill? (This will also help reduce conflicts as we have others working on similar skills that used to, in the C# version, store their prompts in a "constants" file.)
> 
> Open to thoughts/feedback!

Hi Jordan! Happy to help! Some of the prompts are really long and I feel like it's less readable if it's just part of the code file, so I still have them in a constants file for now. Please let me know if you feel strongly the other way and I'll be happy to change it.

**Comment by jjhenkel** [comment]
That sounds good for now @kitnhl! I think we may eventually try and load them from file or something to clean it up, but that can come later. I'll take a look at this PR soon, looks like we may be blocked by some of the partitioning features being missing (I know someone is actively working on that though).

File: python/semantic_kernel/core_skills/conversation_summary_skill.py [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+            skill_name=type(self).__name__,
+            description="Given a section of a conversation transcript, summarize the part of the conversation.",
+            max_tokens=self.MAX_TOKENS,
+            temperature=0.1,
+            top_p=0.5,
+        )
+
+    @sk_function(
+        description="Given a long conversation transcript, identify action items.",
+        name="getConvesationActionItemsAsync",
```
Comment (mkarle) [inline python/semantic_kernel/core_skills/conversation_summary_skill.py:52]: typo, getConversationActionItemsAsync

File: python/semantic_kernel/core_skills/conversation_summary_skill.py [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+        input_description="A long conversation transcript.",
+    )
+    async def get_convesation_action_items_async(
```
Comment (mkarle) [inline python/semantic_kernel/core_skills/conversation_summary_skill.py:55]: typo here too

File: python/semantic_kernel/core_skills/conversation_summary_skill.py [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+
+        # TODO: Uncomment the following code when aggregate_partitioned_results is implemented.
+        # return await self._conversation_action_items_function.aggregate_partitioned_results_async(paragraphs, context)
+
+    @sk_function(
+        description="Given a long conversation topic, identify topics worth remembering.",
+        name="getConvesationTopicsAsync",
+        input_description="A long conversation transcript.",
+    )
+    async def get_convesation_topics_async(
```
Comment (mkarle) [inline python/semantic_kernel/core_skills/conversation_summary_skill.py:77]: typo

**Review by mkarle — CHANGES_REQUESTED** [review]
Just some typos. The rest can be done when unblocked by the partitioner.

**Review by mkarle — APPROVED** [review]

**Pull request merged**